chapter-07/01-sdks/shared/src/configuration.rs

```diff
@@ -0,0 +1,181 @@
+use aws_sdk_ssm::Client;
+use figment::providers::{Env, Format, Json, Serialized};
+use figment::Figment;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+enum LogLevel {
+    TRACE,
+    INFO,
+    WARN,
+    ERROR,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Configuration {
+    pub table_name: String,
+    pub log_level: LogLevel,
+    pub api_key: String,
+}
+
+impl Configuration {
+    pub async fn load(ssm_client: &Client, secret_client: &aws_sdk_secretsmanager::Client) -> Self {
+        let mut config = Figment::from(Serialized::defaults(Configuration::default()))
+            // .merge fills in any missing values from the environment
+            .merge(Env::prefixed("APP_"));
+
+        let ssm_configuration = Configuration::load_from_ssm(ssm_client).await;
+        config = match ssm_configuration {
+            Ok(ssm_config) => config.merge(Json::string(&ssm_config)),
+            Err(_) => config,
+        };
+
+        let secret_manager_configuration =
+            Configuration::load_from_secret_manager(secret_client).await;
+        config = match secret_manager_configuration {
+            Ok(secret_config) => config
+                // .join overrides any existing values with new values from this JSON
+                .join(Json::string(&secret_config)),
+            Err(_) => config,
+        };
+
+        let config = config.extract();
+
+        match config {
+            Ok(config) => {
+                println!("{}", config);
+                config
+            }
+            Err(e) => {
+                eprintln!("Failed to load configuration: {:?}", e);
+                Configuration::default()
+            }
+        }
```

Comment:
I think i mentioned this somewhere else (maybe in the other PR) but I feel this one would be better suited if it returned a `Result<Self, Error>` (where `Error` is probably some kind of config error, maybe something provided by `figment`). 

Just loading the default configuration in case of an error seems risky to me. I would prefer the whole lambda to fail at init-time instead.